phac-nml/irida-next
# Metadata File Import UI

## What does this PR do and why?
Created a new metadata file import modal.
Fixes #357

## Screenshots or screen recordings
![image](https://github.com/phac-nml/irida-next/assets/325703/5cd749c8-c9a8-4b91-b8bf-1fc5ecf95f01)

Successful import:
![image](https://github.com/phac-nml/irida-next/assets/325703/de27353f-de95-4324-a100-6739e09191e9)

File validation errors:
![image](https://github.com/phac-nml/irida-next/assets/325703/250869b2-c9d9-4472-9de8-bdc4cf5f5662)

Missing sample errors:
![image](https://github.com/phac-nml/irida-next/assets/325703/d9631d41-c3fe-4330-a52c-ef41b5eb56f5)

## How to set up and validate locally
1. Navigate to a project samples page that the logged in user has access to update.
2. Hit the `Import Metadata` button.
3. Select a file.
4. Select the `Sample ID Column`. (Note: Currently finds samples by name.)
5. Hit the `Import Metadata` button.
6. Verify the sample metadata was updated.
7. Try with and without ignoring empty values.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
